internal/commands/user/user_state.go

```diff
@@ -0,0 +1,184 @@
+package user
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+
+	"github.com/spf13/pflag"
+)
+
+// NewCommandDisable returns a *CommandUserState which is the `user disable` command
+func NewCommandDisable() *CommandUserState {
+	return &CommandUserState{}
+}
+
+// NewCommandEnable returns a *CommandUserState which is the `user enable` command
+func NewCommandEnable() *CommandUserState {
+	return &CommandUserState{userEnable: true}
+}
+
+// CommandUserState is the `user enable/disable` command
+type CommandUserState struct {
+	userEnable  bool
+	inputs      userStateInputs
+	outputs     []userOutput
+	realmClient realm.Client
+}
+
+type userStateInputs struct {
+	app.ProjectInputs
+	usersInputs
+}
+
+func (i *userStateInputs) Resolve(profile *cli.Profile, ui terminal.UI) error {
+	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Flags is the command flags
+func (cmd *CommandUserState) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+	var flagUsage string
+	if cmd.userEnable {
```

Comment:
after seeing how simple these commands end up being (and how similar they are to your previous `user delete` work), I'm leaning towards ditching the abstraction here all together and just defining/implementing `CommandEnable`, `CommandDisable`, `CommandDelete` as their own structs (with their own inputs).

along that same line, we can probably start to abstract away some of the `Feedback` logic we share across many of these commands.  I think that `[]userOutput` => `map` => `[]terminal.Log` flow is fairly similar across the commands

Reply:
This is a good point that I noticed with user revoke as well. I had a couple of comments [in that PR](https://github.com/10gen/realm-cli/pull/54) pointing out the similarities and am happy to adjust anything there as well

Reply:
I agree with the `Feedback` refactor to be reusable throughout the command set. 
For ditching the abstraction, I'm not quite in alignment. The main thought I have is following DRY and I'm not seeing any red flags from the abstractions made so far to warrant not following DRY.

Reply:
this particular DRY effort/abstraction just is not very go-like, if we were working in a language with better generic support this type of approach would fit a little more nicely